src/msg/async/EventKqueue.h

```diff
@@ -25,18 +25,33 @@
 
 class KqueueDriver : public EventDriver {
   int kqfd;
-  struct kevent *events;
+  pthread_t mythread;
+  struct kevent *res_events;
   CephContext *cct;
   int size;
 
+ private:
+  // Keep what we set on the kqfd
+  struct SaveEvent{
+    int fd;
+    int mask;
+  };
```

Comment:
@yuyuyu101 
That would actually make it hard to use `res_events[]` the structure that `kevent()` can return the triggerd events in. So we'd again need to `alloc(nevents-size)` an array for that, or do that on the stack. Blowing the stack out of proportion on every call to `event_wait()`